src/arith_uint256.h

```diff
@@ -174,7 +174,7 @@ class base_uint
     {
         // prefix operator
         int i = 0;
-        while (++pn[i] == 0 && i < WIDTH-1)
+        while (i < WIDTH && ++pn[i] == 0)
```

Comment:
Do you think adding:
`static_assert(BITS/32 > 0 && BITS%32 == 0, "Template parameter BITS must be a positive multiple of 32."");`
in the class constructor would be fine?